assets/js/src/fade.js

```diff
@@ -1,35 +1,59 @@
-const mobileOffset = 270;
-// Start with nav-work button highlighted and only work section showing
-document.getElementById('nav-work').classList.add('current-btn');
-document.getElementById('work').classList.add('display');
+const headerElement = document.getElementById('header');
+const headerGeometry = headerElement.getBoundingClientRect();
+const stickyNavOffset = headerGeometry.height + window.scrollY + headerGeometry.y;
+const navElements = Array.from(document.getElementsByClassName('nav-btn'));
+const sectionElements = Array.from(document.getElementsByTagName('section'));
+const fadeElement = document.getElementById('fade-wrapper');
 
-const navElements = document.getElementsByClassName('nav-btn');
+function selectWithLocation(el, locationHash) {
+  return el.dataset.section === locationHash.substring(1);
+}
 
-Array.from(navElements).forEach((navElement) => {
-  navElement.addEventListener('click', (e) => {
-    e.preventDefault();
-    Array.from(document.getElementsByTagName('li'))
-      .forEach(el => el.classList.remove('current-btn'));
-    document.getElementById(e.target.id).classList.add('current-btn');
-    $('#fade-wrapper').fadeOut(500, () => {
-      Array.from(document.getElementsByTagName('section'))
-        .forEach(el => el.classList.remove('display'));
-      document.getElementById(e.target.id.slice(4)).classList.add('display');
-      if (window.scrollY > mobileOffset) {
-        window.scrollTo(0, mobileOffset);
-      }
-      $('#fade-wrapper').fadeIn(500);
-    });
+function getActiveElements(locationHash) {
+  if (locationHash && locationHash !== '') {
+    const navElement = navElements.find(el => selectWithLocation(el, locationHash));
+    const sectionElement = sectionElements.find(el => selectWithLocation(el, locationHash));
+    return [navElement, sectionElement];
+  }
+  return [navElements[0], sectionElements[0]];
+}
```

Comment:
Rather than a function which:

1. Knows about navs and sections
2. Returns a tuple

You passed the set of elements as an argument, and called it twice, once for each set?

Reply:
Yes, And that should also let me get rid of my other function `selectWithLocation`